src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -143,6 +200,8 @@ static sd_dhcp_server *dhcp_server_free(sd_dhcp_server *server) {
 
         hashmap_free(server->leases_by_client_id);
 
+        server->vendor_specific = set_free(server->vendor_specific);
```

Comment:
it's stored in Networkd's data not in lib we just passing the pointer. Like the interface is down we free the memory  like server one and again it up we reallocate and init it . If we free it while interface is down then come up it will crash